R/utils.R

```diff
@@ -230,3 +230,68 @@ padding_lst <- function(ft, indents) {
     flextable::padding(ft, s, 1, padding.left = (indents[s] + 1) * 10)
   }, seq_len(length(indents)), ft)
 }
+
+#' Split a text block into smaller blocks with a specified number of lines.
+#'
+#' This function takes a block of text and divides it into smaller blocks, each containing
+#' a specified number of lines.
+#'
+#' @param block_text A character vector containing the input block of text.
+#' @param n The number of lines per block.
+#'
+#' @return A list of character vectors, where each element is a smaller block of text
+#'         containing 'n' lines. If the input block of text has fewer lines than 'n', the
+#'         entire block is returned as a single element list.
+#'
+#' @keywords internal
+split_text_into_blocks <- function(block_text, n) {
+  lines <- strsplit(block_text, "\n")[[1]]
+  num_lines <- length(lines)
+
+  if (num_lines <= n) {
+    return(list(block_text))
+  }
+
+  num_blocks <- ceiling(num_lines / n)
+  blocks <- vector("list", length = num_blocks)
+
+  for (i in 1:num_blocks) {
+    start <- (i - 1) * n + 1
+    end <- min(i * n, num_lines)
+    block <- paste(lines[start:end], collapse = "\n")
+    blocks[[i]] <- block
+  }
+
+  return(blocks)
+}
+
+#' Extract a Field from YAML and Optionally Retrieve Names from a List
+#'
+#' This function parses a YAML text and extracts the specified field. It provides
+#' the option to retrieve the names of elements from a list if the field contains a list.
+#'
+#' @param yaml_text A character vector containing the YAML text.
+#' @param field_name The name of the field to extract.
+#' @param check_list Logical, indicating whether to check if the result is a list
+#'                   and retrieve the names of list elements. Default is TRUE.
+#'
+#' @return If `check_list` is TRUE and the result is a list, it returns the names of
+#'         elements in the list; otherwise, it returns the extracted field.
+#'
+#' @keywords internal
+reverse_yaml_field <- function(yaml_text, field_name, check_list = TRUE) {
```

Comment:
The `check_list` argument is never used, why have it at all?

Reply:
I designed this function while exploring autoslideR and looking for a way to utilize templates. At that time, it seemed like a good idea to include this argument as "logical," which indicates whether to check if the result is a list. If it is, the function will return a complete nested list; otherwise, it will return just the value of the first name.

However, at this point, it is not necessary. I will proceed with removing it.
<img width="705" alt="image" src="https://github.com/insightsengineering/teal.reporter/assets/6700955/8f684227-f649-434a-95d8-d5dddd7f1422">